### ISSUE TITLE
Linker error for application with imageio on 21.1.0.r16. "multiple definition of `jvm';"

### ISSUE THREAD

**Issue by bademux** [oracle/graal#3399]
Build for application with imageio on graalvm 21.1.0.r16 fails with
```
ld: lib/static/linux-amd64/glibc/libawt.a(awt_LoadLibrary.o):(.bss.jvm+0x0): multiple definition of `jvm'; lib/static/linux-amd64/glibc/libjavajpeg.a(jpegdecoder.o):(.bss.jvm+0x0): first defined here
```
On 21.1.0.r11 the build is OK.

**Steps to reproduce the issue**
Please include both build steps as well as run steps
1. ```sdk install java 21.1.0.r16-grl``` https://github.com/graalvm/graalvm-ce-builds/releases/download/vm-21.1.0/graalvm-ce-java16-linux-amd64-21.1.0.tar.gz
2. Compile project with imageio without "--static" option

**Describe GraalVM and your environment:**
 - GraalVM version (latest snapshot builds can be found [here](https://github.com/graalvm/graalvm-ce-dev-builds/releases)), or commit id if built from source: **21.1.0.r16**
 - JDK major version: **16**
 - OS: **Ubuntu 20.04**
 - Architecture: **AMD64**

<details> <summary>full logs</summary>
<pre>
Fatal error:java.lang.RuntimeException: java.lang.RuntimeException: There was an error linking the native image: Linker command exited with 1

Linker command executed:
/usr/bin/gcc -z noexecstack -Wl,--gc-sections -Wl,--dynamic-list -Wl,/tmp/SVM-8474799773218608047/exported_symbols.list -Wl,--exclude-libs,ALL -Wl,-x -o /home/user/workspace/pawgen/build/bin/pawgen pawgen.o /home/user/.sdkman/candidates/java/21.1.0.r16-grl/lib/static/linux-amd64/glibc/libnet.a /home/user/.sdkman/candidates/java/21.1.0.r16-grl/lib/static/linux-amd64/glibc/libjavajpeg.a /home/user/.sdkman/candidates/java/21.1.0.r16-grl/lib/static/linux-amd64/glibc/libextnet.a /home/user/.sdkman/candidates/java/21.1.0.r16-grl/lib/static/linux-amd64/glibc/libnio.a /home/user/.sdkman/candidates/java/21.1.0.r16-grl/lib/svm/clibraries/linux-amd64/liblibchelper.a /home/user/.sdkman/candidates/java/21.1.0.r16-grl/lib/static/linux-amd64/glibc/libjava.a /home/user/.sdkman/candidates/java/21.1.0.r16-grl/lib/static/linux-amd64/glibc/liblcms.a /home/user/.sdkman/candidates/java/21.1.0.r16-grl/lib/static/linux-amd64/glibc/libfontmanager.a /home/user/.sdkman/candidates/java/21.1.0.r16-grl/lib/static/linux-amd64/glibc/libawt_headless.a /home/user/.sdkman/candidates/java/21.1.0.r16-grl/lib/static/linux-amd64/glibc/libawt.a /home/user/.sdkman/candidates/java/21.1.0.r16-grl/lib/static/linux-amd64/glibc/libharfbuzz.a /home/user/.sdkman/candidates/java/21.1.0.r16-grl/lib/static/linux-amd64/glibc/libfdlibm.a /home/user/.sdkman/candidates/java/21.1.0.r16-grl/lib/static/linux-amd64/glibc/libzip.a /home/user/.sdkman/candidates/java/21.1.0.r16-grl/lib/svm/clibraries/linux-amd64/libjvm.a -v -L/tmp/SVM-8474799773218608047 -L/home/user/.sdkman/candidates/java/21.1.0.r16-grl/lib/static/linux-amd64/glibc -L/home/user/.sdkman/candidates/java/21.1.0.r16-grl/lib/svm/clibraries/linux-amd64 -lstdc++ -lm -lfreetype -lpthread -ldl -lz -lrt

Linker command output:
Using built-in specs.
COLLECT_GCC=/usr/bin/gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/9/lto-wrapper
OFFLOAD_TARGET_NAMES=nvptx-none:hsa
OFFLOAD_TARGET_DEFAULT=1
Target: x86_64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Ubuntu 9.3.0-17ubuntu1~20.04' --with-bugurl=file:///usr/share/doc/gcc-9/README.Bugs --enable-languages=c,ada,c++,go,brig,d,fortran,objc,obj-c++,gm2 --prefix=/usr --with-gcc-major-version-only --program-suffix=-9 --program-prefix=x86_64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-plugin --enable-default-pie --with-system-zlib --with-target-system-zlib=auto --enable-objc-gc=auto --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-offload-targets=nvptx-none=/build/gcc-9-HskZEa/gcc-9-9.3.0/debian/tmp-nvptx/usr,hsa --without-cuda-driver --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
Thread model: posix
gcc version 9.3.0 (Ubuntu 9.3.0-17ubuntu1~20.04) 
COMPILER_PATH=/usr/lib/gcc/x86_64-linux-gnu/9/:/usr/lib/gcc/x86_64-linux-gnu/9/:/usr/lib/gcc/x86_64-linux-gnu/:/usr/lib/gcc/x86_64-linux-gnu/9/:/usr/lib/gcc/x86_64-linux-gnu/
LIBRARY_PATH=/usr/lib/gcc/x86_64-linux-gnu/9/:/usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/:/usr/lib/gcc/x86_64-linux-gnu/9/../../../../lib/:/lib/x86_64-linux-gnu/:/lib/../lib/:/usr/lib/x86_64-linux-gnu/:/usr/lib/../lib/:/usr/lib/gcc/x86_64-linux-gnu/9/../../../:/lib/:/usr/lib/
COLLECT_GCC_OPTIONS='-z' 'noexecstack' '-o' '/home/user/workspace/pawgen/build/bin/pawgen' '-v' '-L/tmp/SVM-8474799773218608047' '-L/home/user/.sdkman/candidates/java/21.1.0.r16-grl/lib/static/linux-amd64/glibc' '-L/home/user/.sdkman/candidates/java/21.1.0.r16-grl/lib/svm/clibraries/linux-amd64' '-mtune=generic' '-march=x86-64'
 /usr/lib/gcc/x86_64-linux-gnu/9/collect2 -plugin /usr/lib/gcc/x86_64-linux-gnu/9/liblto_plugin.so -plugin-opt=/usr/lib/gcc/x86_64-linux-gnu/9/lto-wrapper -plugin-opt=-fresolution=/tmp/ccWyTN9X.res -plugin-opt=-pass-through=-lgcc -plugin-opt=-pass-through=-lgcc_s -plugin-opt=-pass-through=-lc -plugin-opt=-pass-through=-lgcc -plugin-opt=-pass-through=-lgcc_s --build-id --eh-frame-hdr -m elf_x86_64 --hash-style=gnu --as-needed -dynamic-linker /lib64/ld-linux-x86-64.so.2 -pie -z now -z relro -o /home/user/workspace/pawgen/build/bin/pawgen -z noexecstack /usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/Scrt1.o /usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/crti.o /usr/lib/gcc/x86_64-linux-gnu/9/crtbeginS.o -L/tmp/SVM-8474799773218608047 -L/home/user/.sdkman/candidates/java/21.1.0.r16-grl/lib/static/linux-amd64/glibc -L/home/user/.sdkman/candidates/java/21.1.0.r16-grl/lib/svm/clibraries/linux-amd64 -L/usr/lib/gcc/x86_64-linux-gnu/9 -L/usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu -L/usr/lib/gcc/x86_64-linux-gnu/9/../../../../lib -L/lib/x86_64-linux-gnu -L/lib/../lib -L/usr/lib/x86_64-linux-gnu -L/usr/lib/../lib -L/usr/lib/gcc/x86_64-linux-gnu/9/../../.. --gc-sections --dynamic-list /tmp/SVM-8474799773218608047/exported_symbols.list --exclude-libs ALL -x pawgen.o /home/user/.sdkman/candidates/java/21.1.0.r16-grl/lib/static/linux-amd64/glibc/libnet.a /home/user/.sdkman/candidates/java/21.1.0.r16-grl/lib/static/linux-amd64/glibc/libjavajpeg.a /home/user/.sdkman/candidates/java/21.1.0.r16-grl/lib/static/linux-amd64/glibc/libextnet.a /home/user/.sdkman/candidates/java/21.1.0.r16-grl/lib/static/linux-amd64/glibc/libnio.a /home/user/.sdkman/candidates/java/21.1.0.r16-grl/lib/svm/clibraries/linux-amd64/liblibchelper.a /home/user/.sdkman/candidates/java/21.1.0.r16-grl/lib/static/linux-amd64/glibc/libjava.a /home/user/.sdkman/candidates/java/21.1.0.r16-grl/lib/static/linux-amd64/glibc/liblcms.a /home/user/.sdkman/candidates/java/21.1.0.r16-grl/lib/static/linux-amd64/glibc/libfontmanager.a /home/user/.sdkman/candidates/java/21.1.0.r16-grl/lib/static/linux-amd64/glibc/libawt_headless.a /home/user/.sdkman/candidates/java/21.1.0.r16-grl/lib/static/linux-amd64/glibc/libawt.a /home/user/.sdkman/candidates/java/21.1.0.r16-grl/lib/static/linux-amd64/glibc/libharfbuzz.a /home/user/.sdkman/candidates/java/21.1.0.r16-grl/lib/static/linux-amd64/glibc/libfdlibm.a /home/user/.sdkman/candidates/java/21.1.0.r16-grl/lib/static/linux-amd64/glibc/libzip.a /home/user/.sdkman/candidates/java/21.1.0.r16-grl/lib/svm/clibraries/linux-amd64/libjvm.a -lstdc++ -lm -lfreetype -lpthread -ldl -lz -lrt -lgcc --push-state --as-needed -lgcc_s --pop-state -lc -lgcc --push-state --as-needed -lgcc_s --pop-state /usr/lib/gcc/x86_64-linux-gnu/9/crtendS.o /usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/crtn.o
/usr/bin/ld: /home/user/.sdkman/candidates/java/21.1.0.r16-grl/lib/static/linux-amd64/glibc/libawt.a(awt_LoadLibrary.o):(.bss.jvm+0x0): multiple definition of `jvm'; /home/user/.sdkman/candidates/java/21.1.0.r16-grl/lib/static/linux-amd64/glibc/libjavajpeg.a(jpegdecoder.o):(.bss.jvm+0x0): first defined here
collect2: error: ld returned 1 exit status
</pre>
</details>

**Comment by oubidar-Abderrahim** [comment]
Hi, do you have an example that we can use to reproduce this issue? please share it, thank you.

**Comment by bademux** [comment]
Hi, I'm trying to compile https://github.com/bademux/pawgen

**Comment by bademux** [comment]
closed as "clone" in favour of #3938
